src/arkit.cs

```diff
@@ -0,0 +1,493 @@
+	}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[BaseType (typeof(NSObject))]
+	interface ARSession
+	{
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		IARSessionDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegateQueue", ArgumentSemantic.Strong)]
+		DispatchQueue DelegateQueue { get; set; }
+
+		[NullAllowed, Export ("currentFrame", ArgumentSemantic.Copy)]
+		ARFrame CurrentFrame { get; }
+
+		[NullAllowed, Export ("configuration", ArgumentSemantic.Copy)]
+		ARSessionConfiguration Configuration { get; }
+
+		[Export ("runWithConfiguration:")]
+		void Run (ARSessionConfiguration configuration);
+
+		[Export ("runWithConfiguration:options:")]
+		void Run (ARSessionConfiguration configuration, ARSessionRunOptions options);
+
+		[Export ("pause")]
+		void Pause ();
+
+		[Export ("addAnchor:")]
+		void AddAnchor (ARAnchor anchor);
+
+		[Export ("removeAnchor:")]
+		void RemoveAnchor (ARAnchor anchor);
+	}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[Protocol]
+	[BaseType (typeof(NSObject))]
+	interface ARSessionObserver
+	{
+		[Export ("session:didFailWithError:")]
+		void DidFail (ARSession session, NSError error);
+
+		[Export ("session:cameraDidChangeTrackingState:")]
+		void CameraDidChangeTrackingState (ARSession session, ARCamera camera);
+
+		[Export ("sessionWasInterrupted:")]
+		void SessionWasInterrupted (ARSession session);
+
+		[Export ("sessionInterruptionEnded:")]
+		void SessionInterruptionEnded (ARSession session);
+	}
+
+	interface IARSessionDelegate {}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject))]
+	interface ARSessionDelegate : ARSessionObserver
+	{
+		[Export ("session:didUpdateFrame:")]
+		void DidUpdateFrame (ARSession session, ARFrame frame);
+
+		[Export ("session:didAddAnchors:")]
+		void DidAddAnchors (ARSession session, ARAnchor[] anchors);
+
+		[Export ("session:didUpdateAnchors:")]
+		void DidUpdateAnchors (ARSession session, ARAnchor[] anchors);
+
+		[Export ("session:didRemoveAnchors:")]
+		void DidRemoveAnchors (ARSession session, ARAnchor[] anchors);
+	}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[BaseType (typeof(NSObject))]
+	interface ARSessionConfiguration : NSCopying
+	{
+		[Static]
+		[Export ("isSupported")]
+		bool IsSupported { get; }
+
+		[Export ("worldAlignment", ArgumentSemantic.Assign)]
+		ARWorldAlignment WorldAlignment { get; set; }
+
+		[Export ("lightEstimationEnabled")]
+		bool LightEstimationEnabled { [Bind ("isLightEstimationEnabled")] get; set; }
+	}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[BaseType (typeof(ARSessionConfiguration))]
+	interface ARWorldTrackingSessionConfiguration
+	{
+		[Export ("planeDetection", ArgumentSemantic.Assign)]
+		ARPlaneDetection PlaneDetection { get; set; }
+	}
+
+	[iOS (11,0)]
+	[NoWatch, NoTV, NoMac]
+	[Static]
+	interface ARSCNDebugOptions
+	{
+		[Internal]
+		[Field ("ARSCNDebugOptionShowWorldOrigin")]
+		int _ARSCNDebugOptionShowWorldOrigin { get; }
```

Comment:
@rolfbjarne Here's the enum signature: `public enum SCNDebugOptions : nuint`

If I use `nuint` instead of `int` I'm getting: `error BI1014: bgen: Unsupported type for Fields: global::System.UInt32`

If I use `nint` instead of `int` I'm getting: `ARKit/ARSCNDebugOptions.g.cs(53,65): error CS0030: Cannot convert type 'System.nint' to 'SceneKit.SCNDebugOptions'`